packages/shorebird_cli/lib/src/shorebird_flavor_mixin.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:platform/platform.dart';
+import 'package:shorebird_cli/src/command.dart';
+
+/// Mixin on [ShorebirdCommand] which exposes methods for extracting
+/// product flavors from the current app.
+mixin ShorebirdFlavorMixin on ShorebirdCommand {
+  /// Return the set of product flavors configured for the app at [path].
+  /// Returns an empty set for apps that do not use product flavors.
+  Future<Set<String>> extractProductFlavors(
+    String path, {
+    Platform platform = const LocalPlatform(),
+  }) async {
+    final executable = platform.isWindows ? 'gradlew.bat' : 'gradlew';
+    final androidStudioPath = _getAndroidStudioPath(platform);
+    final javaPath = androidStudioPath != null
+        ? _getJavaPath(androidStudioPath, platform)
+        : null;
+    final result = await process.run(
+      p.join(path, 'android', executable),
+      ['app:tasks', '--all', '--console=auto'],
+      runInShell: true,
+      workingDirectory: p.join(path, 'android'),
+      environment: {
+        if (javaPath != null) 'JAVA_HOME': javaPath,
+      },
+    );
+
+    if (result.exitCode != 0) throw Exception('${result.stderr}');
```

Comment:
When this errored on Windows, the output was simply

```
$ shorebird init
Exception:
```

We should probably include the command in the error output (or find out where Windows is writing to if not stderr)

Reply:
I updated the error to include both `stdout` and `stderr` to hopefully provide more useful output.